c_glib/gandiva-glib/node.cpp

```diff
@@ -395,42 +434,1438 @@ ggandiva_function_node_get_parameters(GGandivaFunctionNode *node)
-                               const gchar *name,
-                               GList *parameters,
-                               GArrowDataType *return_type)
+
+typedef struct GGandivaBooleanLiteralNodePrivate_ {
+  gboolean is_true;
+} GGandivaBooleanLiteralNodePrivate;
+
+enum {
+  PROP_IS_TRUE = 1
+};
+
+G_DEFINE_TYPE_WITH_PRIVATE(GGandivaBooleanLiteralNode,
+                           ggandiva_boolean_literal_node,
+                           GGANDIVA_TYPE_LITERAL_NODE)
+
+#define GGANDIVA_BOOLEAN_LITERAL_NODE_GET_PRIVATE(object)               \
+  static_cast<GGandivaBooleanLiteralNodePrivate *>(                     \
+    ggandiva_boolean_literal_node_get_instance_private(                 \
+      GGANDIVA_BOOLEAN_LITERAL_NODE(object)))
+
+static void
+ggandiva_boolean_literal_node_set_property(GObject *object,
+                                           guint prop_id,
+                                           const GValue *value,
+                                           GParamSpec *pspec)
 {
-  auto function_node = g_object_new(GGANDIVA_TYPE_FUNCTION_NODE,
-                                    "node", gandiva_node,
-                                    "name", name,
-                                    "return-type", return_type,
-                                    NULL);
-  auto priv = GGANDIVA_FUNCTION_NODE_GET_PRIVATE(function_node);
-  for (auto node = parameters; node; node = g_list_next(node)) {
-    auto parameter = GGANDIVA_NODE(node->data);
-    priv->parameters = g_list_prepend(priv->parameters, g_object_ref(parameter));
+  auto priv = GGANDIVA_BOOLEAN_LITERAL_NODE_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_IS_TRUE:
+    priv->is_true = g_value_get_boolean(value);
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
   }
-  priv->parameters = g_list_reverse(priv->parameters);
-  return GGANDIVA_FUNCTION_NODE(function_node);
+}
+
+static void
+ggandiva_boolean_literal_node_get_property(GObject *object,
+                                           guint prop_id,
+                                           GValue *value,
+                                           GParamSpec *pspec)
+{
+  auto priv = GGANDIVA_BOOLEAN_LITERAL_NODE_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_IS_TRUE:
+    g_value_set_boolean(value, priv->is_true);
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+ggandiva_boolean_literal_node_init(GGandivaBooleanLiteralNode *boolean_literal_node)
+{
+}
+
+static void
+ggandiva_boolean_literal_node_class_init(GGandivaBooleanLiteralNodeClass *klass)
+{
+  auto gobject_class = G_OBJECT_CLASS(klass);
+
+  gobject_class->set_property = ggandiva_boolean_literal_node_set_property;
+  gobject_class->get_property = ggandiva_boolean_literal_node_get_property;
+
+  GParamSpec *spec;
+  spec = g_param_spec_boolean("is-true",
```

Comment:
`"value"` is better.
If we use `value` as property name, users can always use `value` property name to get value from all literal nodes.

Reply:
OK. I'll change it.

Reply:
I've fixed it.